Sources/NIOFileSystem/FileSystemProtocol.swift

```diff
@@ -599,7 +605,29 @@ extension FileSystemProtocol {
     public func removeItem(
         at path: FilePath
     ) async throws -> Int {
-        try await self.removeItem(at: path, recursively: true)
+        try await self.removeItem(at: path, strategy: .platformDefault, recursively: true)
+    }
+
+    /// Deletes the file or directory (and its contents) at `path`.
+    ///
+    /// The item to be removed must be a regular file, symbolic link or directory. If no file exists
+    /// at the given path then this function returns zero.
+    ///
+    /// If the item at the `path` is a directory then the contents of all of its subdirectories will
+    /// be removed recursively before the directory at `path`. Symbolic links are removed (but their
+    /// targets are not deleted).
+    ///
+    /// - Parameters:
+    ///   - path: The path to delete.
+    ///   - removalStrategy: Whether to delete files sequentially (one-by-one), or perform a
+    ///       concurrent scan of the tree at `path` and delete files when they are found.
+    /// - Returns: The number of deleted items which may be zero if `path` did not exist.
+    @discardableResult
+    public func removeItem(
+        at path: FilePath,
+        strategy removalStrategy: RemovalStrategy
+    ) async throws -> Int {
+        try await self.removeItem(at: path, strategy: removalStrategy, recursively: true)
```

Comment:
We want the strategy and the recursion behaviour to both be optional. At the moment it's not possible to call `removeItem(at: "foo", recursive: true)` as the strategy isn't defaulted.

Annoyingly I think we need have the following:

- `removeItem(at:strategy:recursively:)` as the requirement, and
- `removeItem(at:)` and
- `removeItem(at:strategy:)`, (defaults recursively) and
- `removeItem(at:recursively:)` (defaults strategy) as extensions.

Reply:
Having a hard time replicating the API breakage failures locally, and my CI-based-debugging is not the most efficient here. Reached out to Rick for some guidance.

I think I've done as you've suggested (`removeItem(at:strategy:recursively:)` as the only requirement, and the other three as extensions to `FileSystemProtocol`), but I'm still seeing issues. Don't I also need an extension for `removeItem(at:strategy:recursively:)` to define a concrete implementation of that variant?

Reply:
Fixed this at last. Thanks @rnro for the pointers!